### DISCUSSION TITLE
Welcome to sbp Discussions!

### DISCUSSION THREAD

**Discussion by hank-cp** [hank-cp/sbp#30] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dtrouillet** [comment]
Hello my name is Damien,
I'm Technical Leader at Informatique CDC, and I love Domotic ( HomeAssistant ❤️  ).
I use SBP for aside project in my company to make a Portal Dev with extendable feature.

I want to participate in this community to share my knowledge and my time 😄  .

Sorry for my bad english :s.

Damien

**Reply by hank-cp** [reply]
Hi Damien,

Welcome again! Right now the SBP core and starter are almost reach their original design goals. The [roadmap](https://github.com/hank-cp/sbp/blob/master/docs/roadmap.md) docs records some of my thought for the next steps of prospect. Do you have any idea in your mind that how sbp could help people better, or any todo items in roadmap you are interested in?

Hank

**Comment by mrcharleshu** [comment]
How to keep update with SpringBoot? current spring boot version is 2.2.0😅

**Reply by hank-cp** [reply]
@mrcharleshu SpringBoot is now supports from 2.1-2.7(today) now.

**Comment by hank-cp** [comment]
@mrcharleshu I tried to upgraded to 2.6, and there are so many errors😭😭. I'll need some time to fix them. 

Another thing, if it's necessary to support different spring boot major versions, it looks necessary to build different sbp branch release to correspending spring boot major versions. That's a lot of works too...

**Comment by hank-cp** [comment]
@mrcharleshu I have create a branch [sprint-boot-2.6](https://github.com/hank-cp/sbp/tree/sprint-boot-2.6), and fix all migrations.
As mentioned in last post, I still need some effort to break down Spring Boot dependencies to more looser level to make sbp keeps up with Sprint Boot version updates.

For the moment, If you stilll want to use sbp to latest Spring boot version, you could chack out this repo, then build and publish jars to your local repository.

**Comment by Nandu85** [comment]
Hello, @hank-cp I am new to this concept of making plugins and adding it to container. I have checked concept enable and disable plugin. I found that it is handled using text files. 
But what, if we want to handle it using database. Like we can make a collection of plugin and store plugin-id and status in it. Then we use it for further operations. Here I am thinking to store status enabled and disabled in it. 
Please share your opinion for the same.

**Reply by hank-cp** [reply]
sbp's plugin management feature comes from pf4j, which only provide filesystem based plugin loading.

Personally I also think filesystem based plugin loading system is good enough for monolithic service. The text file you mentions is actually the meta-infomation of plugin, which should be shipped with the plugins jar/zip together. It's always fined to choose other place to store plugin, [pf4j also provides a maven repository integration example](https://github.com/pf4j/pf4j-update). But after all, plugins are still downloaded to filesystem and loaded from there.

**Comment by lanjishi** [comment]
hello,my name is lanjishi 
I have a trouble，I have a plugin use springboot and thymeleaf ,it can run himself，bug when it load in the sbpapp, it can not find the html file , js file,and css file which in the plugin resources folder. how can I solve it

**Reply by hank-cp** [reply]
Sbp should have supported static file resource. could you create a new issue for this and throw more detail infomation, like server log or a reproducible repo.

**Reply by kenpb** [reply]
Hi, I've also had problems getting Thymeleaf to read the plugin templates but got it working on https://github.com/kenpb/sbp_demo/tree/thymeleaf though, I'm not sure this was the right way to do it, can you take a look? (Notice that I haven't tested it in "prod mode")

I also have a "working" maven setup for the JAR prod mode creation based on the pf4j tutorial in the same project/branch, maybe I can PR the docs to expand about this? It doesn't feel right since I don't do anything with the plugin.properties file, any thoughts?

**Reply by hank-cp** [reply]
hi @kenpb, I've checked out your repo and everything looks ok. Do you mind creating a PR to add `sbp-spring-boot-starter-thymeleaf` to sbp project? 

Thanks for the nice job:)

**Reply by kenpb** [reply]
@hank-cp Sure thing, I'll get to it this weekend

**Comment by NpkNarola** [comment]
Hi, @hank-cp , We have added sbp plugin management system in our project, but we are facing issues in that. It is, when we upgraded spring webflux project libraries to newer spring boot version "3.2.4".
With this upgradation, plugin is also able to start and list plugin endpoint is also working, but the endpoints to start and stop plugin are giving error in response like, "Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag."
So, I was just in doubt that, does sbp 3.0 snapshot version support current spring boot version? And if it supports, how can I resolve this error?

**Reply by hank-cp** [reply]
Haven't test with spring boot 3.2.x. Will do it in next few days.

Have you test with '-parameters' compiler flag?

**Reply by NpkNarola** [reply]
Actually, we are using your spring boot starter dependency. I am not calling gradle command here manually. I think, it is called internally from library for starting a plugin. I am just hitting plugin start API.
And yes for your information, we have also upgrade a gradle version to 8.5.

**Reply by hank-cp** [reply]
Hi @NpkNarola, this should be a SpringBoot 3.2 migration issue. Check [this](https://stackoverflow.com/questions/77635974/error-name-for-argument-of-type-java-util-uuid-not-specified-and-parameter-n) out.
